prime-router/src/main/kotlin/history/azure/DeliveryFunction.kt

```diff
@@ -45,9 +47,13 @@ class DeliveryFunction(
      * @return Name for the organization
      */
     override fun getOrgName(organization: String): String? {
-        val receiver = workflowEngine.settings.findReceiver(organization)
-        receivingOrgSvc = receiver?.name
-        return receiver?.organizationName
+        return if (organization.contains(Sender.fullNameSeparator)) {
+            workflowEngine.settings.findReceiver(organization).also { receivingOrgSvc = it?.name }
+        } else {
+            workflowEngine.settings.receivers.find {
```

Comment:
Out of curiosity, why are we checking for active status here and not [in SubmissionFunction](https://github.com/CDCgov/prime-reportstream/pull/8332/files#diff-c78040f89c39c1a4b5c02d72cfb1f1430853317a722a5f299f4eecb56dc0b405R41)?  Senders and Receivers both should be able to be disabled, no?

Reply:
The reason I'm asking is -- and this is definitely outside the scope of this pull request -- if the logic could be the same for both types of services, it feels like there's an opportunity for some consolidation here, just bifurcated by service type.  Basically:

- Find the service (Sender|Receiver) by Organization name
- Check if there's a period separator
    - If there is, store the latter half as an instance variable
    - If there isn't find the name of the service by type + Organization name

As far as I can tell, it looks like there's just [some fallback logic](https://github.com/CDCgov/prime-reportstream/blob/master/prime-router/src/main/kotlin/Sender.kt#L206-L213) for Senders if there aren't two fragments in the specified Organization name whereas Receivers [would error](https://github.com/CDCgov/prime-reportstream/blob/master/prime-router/src/main/kotlin/Receiver.kt#L272-L278).

What are your thoughts?

Reply:
Thanks for the insightful questions!

All great observations, I share your desire to consolidate. This went through several revisions and I settled on treading lightly because:
- other approaches impacted functions used elsewhere in the project
- the classes mentioned are due for some refactoring:
  - `getOrgName` is actually a validator/getter (and that logic likely belongs elsewhere as you mentioned)
  - there's a strong case for a mix-in design pattern here to avoid some unnecessary OO pain
- as you mentioned, it felt out of scope for an already scope++ issue
- one more reason my neurons are playing hard-to-get with right now

> Out of curiosity, why are we checking for active status here and not [in SubmissionFunction](https://github.com/CDCgov/prime-reportstream/pull/8332/files#diff-c78040f89c39c1a4b5c02d72cfb1f1430853317a722a5f299f4eecb56dc0b405R41)? Senders and Receivers both should be able to be disabled, no?

In SubmissionFunction, we always have a fully qualified org name to retrieve the Sender (because `default` is always a valid Sender for an org). Following that callstack, we call `findSender()` -> `findSetting()` and the filter IS_ACTIVE on [699](https://github.com/CDCgov/prime-reportstream/blob/master/prime-router/src/main/kotlin/azure/DatabaseAccess.kt#L699).

In DeliveryFunction there is no equivalent fallback, so we need to peruse the list of Receivers which includes ACTIVE and INACTIVE customers. With that said, this comment may have been buried:

> **We may want to change the filtering here (or get last instance instead of first)**

I'm not sure that filter (or any at all) is appropriate, but intent was to match behaviors and only look at active Settings. Regardless, in this logic branch we are not filtering by sending service, so the sender retrieved is irrelevant as it's just used to validate the organization name (as long as it doesn't refer to an out-of-date orgname).

I will add a TODO and create a ticket for a refactor here that will cover consolidating this logic and ensuring no breakage in impacted callers.

Reply:
I agree with Stephen. My concern is if we create new tickets the work described above will never get done. If you are going to push this work, then you'll need to have created and linked the new tickets before approval.

Reply:
I have created a ticket and added it to the TODO for the refactoring / consolidating work.
It still needs grooming, will be done today.